### PR TITLE
Improve tests and attach them to tutorial containers.

### PR DESCRIPTION
**Pull request by MilBia**
A little change for tests construction to partly change them in PBS case. That for make our tests more flexible and should be able to run anywhere. Reason for that was issued I approached during #65 where enough big physical node contain two virtual nodes for PBS cluster, witch make impossible check creation of two nodes, the way we do it so far.
Also according to [comment](https://github.com/E-CAM/jobqueue_features/pull/65#issuecomment-660009779) I add tests for tutorial containers.

### CODE REVIEW

**Comment by ocaisa** [comment]
@MilBia It seems to be getting stuck in the PBS tutorial check, it stops producing output after:
```
jobqueue_features/jobqueue_features/tests/test_pbs_ci.py::TestCIPBS::test_multi_mpi_tasks Found mpirun so assuming we have PBS, running MPI test (with {'implementation': 'standard', 'launcher': 'mpiexec'})
PASSED
```

**Comment by ocaisa** [comment]
If I try it locally, it also seems to time out on the PBS tests (for the tutorial), the job gets past 4 minutes, gets killed and another one does not replace it.

I think this might be hard to solve, I might even consider dropping PBS from the tutorial, we don't really need two options and the PBS one is not really set up properly in general (you can't look at the job output files or anything, you are root in the container, jobs seems to take ages to exit,...)

**Comment by MilBia** [comment]
When I try it local it pass, so I wonder why it's stuck there.

I think, I already make can it works for not root user in PBS container. I need to check before I commit that.
About output files, there is kind a equivalent in command `tracejob n` where `n` is job_id. But yes it's may not be such good as slurm output file.
For that moment I only have no idea about that exit times, which are also really long for me.

**Comment by ocaisa** [comment]
I think the long exit times are because it can't write the output files (see https://community.openpbs.org/t/jobs-stuck-in-exiting-state/283/2). I really think we should just drop PBS for the tutorial part, for training purposes, we don't need to support multiple schedulers.

**Comment by MilBia** [comment]
You're right it probably was problem with permissions in case of stacking those test.
To make exit faster, I require 1GB of memory instead of 2 GB and it's work for me pretty smooth now.
If those don't work for here, I agree to just cut off PBS from tutorial for now.
I only need to add SLURM no root user, as you asked and will commit it.

**Comment by ocaisa** [comment]
@MilBia We can give it an hour but it doesn't look like it will pass

**Comment by MilBia** [comment]
I truly have no idea what going on, CI checking now for PBS script before last commit. I will gave it a minute and probably kill it. I will back to that tomorrow.

**Comment by ocaisa** [comment]
@MilBia I know you have put a lot of effort into that but I really don't think it is worth spending more time on it. The tutorial is about `jobqueue_features`, the resource manager we use is not so relevant. Rather than spend more time on it, I would drop the PBS option from the tutorials and focus only on slurm (which is better behaved anyway)

**Comment by MilBia** [comment]
@ocaisa OK, I will tomorrow remove PBS from tutorial. I'm start thinking it's some kind CI resources or PBS scheduling problem.
Back to SLURM and non root user, I came to the point where I create user, slurm user, and slurm account and try to start some task and it's not go to queue, probably some permission problem.

**Comment by MilBia** [comment]
It's pass CI now, but needed a restart, because build(slurm) stuck on setup step. I leattle losing trust for that gitlab CI.
@ocaisa Maybe I should give PBS another try with travis? Maybe not now, but when we have more time?

File: tutorial/jupyter.sh [hunk 1]
```diff
@@ -12,7 +12,11 @@ function start_slurm() {
     docker exec slurmctld /bin/bash -c "cd /jobqueue_features; pip install -r requirements.txt; pip install --no-deps -e ."
     docker exec c1 /bin/bash -c "cd /jobqueue_features; pip install -r requirements.txt; pip install --no-deps -e ."
     docker exec c2 /bin/bash -c "cd /jobqueue_features; pip install -r requirements.txt; pip install --no-deps -e ."
-    docker exec slurmctld /bin/bash -c "jupyter notebook --ip=0.0.0.0 --port=8888 --allow-root --NotebookApp.token='' --NotebookApp.password='' --NotebookApp.notebook_dir='/jobqueue_features/tutorial'&"
+    docker exec slurmctld /bin/bash -c "adduser slurmuser; chown -R slurmuser /jobqueue_features;"
+    docker exec c1 /bin/bash -c "adduser slurmuser; chown -R slurmuser /jobqueue_features;"
+    docker exec c2 /bin/bash -c "adduser slurmuser; chown -R slurmuser /jobqueue_features;"
+    docker exec slurmctld /bin/bash -c "yes|sacctmgr create account srumuser; yes | sacctmgr create user name=slurmuser Account=srumuser"
```
Comment (ocaisa) [inline tutorial/jupyter.sh]: Typos here for `slurmuser`?

**Review by ocaisa — CHANGES_REQUESTED** [review]
Looks good to me apart from the typo (which I think doesn't matter anyway)

File: tutorial/jupyter.sh (hunk 1 above)
Comment (MilBia) [reply tutorial/jupyter.sh]: Yes. Non root user task don't want run on those containers if the same user (here `slurmuser`) are not on each node.

**Pull request merged**